client_integration_test.go

```diff
@@ -0,0 +1,206 @@
+package notify_test
+
+import (
+	"os"
+	"testing"
+
+	notify "github.com/govau/notify-client-go"
+)
+
+func setup(t *testing.T) (client *notify.Client, smsTemplateID string, emailTemplateID string) {
+	apiKey := os.Getenv("API_KEY")
+	if apiKey == "" {
+		t.Fatal("could not set API_KEY environment variable")
+	}
+
+	smsTemplateID = os.Getenv("SMS_TEMPLATE_ID")
+	if smsTemplateID == "" {
+		t.Fatal("could not set SMS_TEMPLATE_ID environment variable")
+	}
+
+	emailTemplateID = os.Getenv("EMAIL_TEMPLATE_ID")
+	if emailTemplateID == "" {
+		t.Fatal("could not set EMAIL_TEMPLATE_ID environment variable")
+	}
+
+	var err error
+	client, err = notify.NewClient(
+		apiKey,
+	)
+	if err != nil {
+		t.Fatal("could not create Notify client", err)
+	}
+	return
+}
+
+func TestSendSMS(t *testing.T) {
+	client, smsTemplateID, _ := setup(t)
+	phoneNumber := os.Getenv("SMS_RECIPIENT_NUMBER")
+	if phoneNumber == "" {
+		t.Fatal("could not set SMS_RECIPIENT_NUMBER environment variable")
+	}
+
+	ref := "TestSendSMS"
+	resp, err := client.SendSMS(
+		smsTemplateID,
+		phoneNumber,
+		notify.Personalisation{
+			{"name", "John"},
+			{"day", "Friday"},
+		},
+		notify.Reference(ref),
+	)
+	if err != nil {
+		t.Errorf("could not send SMS: %v", err)
```

Comment:
@andrewoh531 you do probably want to keep this as `t.Fatalf` so that the test exists here. Generally speaking you want the test to exit whenever there's an error, unless your test is concerned with checking what the error was and wanting to continue with other checks.